indirect_with_allocators.h

```diff
@@ -0,0 +1,106 @@
+#ifndef XYZ_INDIRECT_WITH_ALLOCATORS_H
+#define XYZ_INDIRECT_WITH_ALLOCATORS_H
+#include <memory>
+#include <utility>
+
+namespace xyz {
+template <class T, class A = std::allocator<T>>
```

Comment:
Presumably `indirect` should now have a `get_allocator` method like other container types?